code/modules/xenoarchaeology/traits/traits.dm

```diff
@@ -0,0 +1,260 @@
+/*
+	For laternate versions of traits, that you want as seperate on the labeler, use the format-
+	'trait_name Δ' 'trait_name Σ' 'trait_name Ω', for up to three alternates, add more symbols if you have a trait with 4 alts or more
+*/
+
+/datum/xenoartifact_trait
+	///Reference to the artifact
+	var/datum/component/xenoartifact/parent
+
+	///Acts as a descriptor for when examining - 'reinforced' 'electrified' 'hollow'
+	var/material_desc
+	///Used when labeler needs a name and trait is too sneaky to have a descriptor when examining.
+	var/label_name
+	///Alternate name displayed when hovering
+	var/alt_label_name
+	///Something briefly explaining it in inagame terms.
+	var/label_desc
+
+	///Asscoiated flags for artifact typing and such
+	var/flags = XENOA_BLUESPACE_TRAIT | XENOA_PLASMA_TRAIT | XENOA_URANIUM_TRAIT | XENOA_BANANIUM_TRAIT | XENOA_PEARL_TRAIT
+	///Other traits this trait wont work with.
+	var/list/blacklist_traits = list()
+	///How rare is this trait? 100 being common, and 1 being very rare
+	var/rarity = XENOA_TRAIT_WEIGHT_COMMON
+
+	///Does this trait reigster targets?
+	var/register_targets = TRUE
+
+	///How much time does this trait add to the artifact cooldownm
+	var/cooldown = 0 SECONDS
+
+	///What trait priority we use
+	var/priority = TRAIT_PRIORITY_ACTIVATOR
+
+	///List of things we've effected. used to automatically reigster & unregister targets. Don't confuse with parent targets, which is things we want to effect
+	var/list/targets = list()
+	///A distinct list of targets, incorporating overrides
+	var/list/focus = list()
+	///Extra target range we add to the artifact
+	var/extra_target_range = 0
+
+	///How much extra value does this trait apply to the artifact - It's important this is applied before anyone can use stickers on the artifact
+	var/extra_value = 0
+	///How many discovery points does this trait give?
+	var/discovery_reward = 100
+
+	///Does this trait contribute to calibration
+	var/contribute_calibration = TRUE
+
+	///Can this trait be made a pearl? - aka can this trait be used in circuits
+	var/can_pearl = TRUE
+
+	///Characteristics for deduction
+	var/weight = 1 //KG
+	var/conductivity = 1 //microsiemens per centimeter - I had to look this up - Don't worry about making this accurate / reasonable
+
+	///What kind of incompatabilities does this trait have
+	var/incompatabilities
+
+/datum/xenoartifact_trait/New(atom/_parent)
+	. = ..()
+	if(_parent)
+		register_parent(_parent)
```

Comment:
What of parent? component parent? parent of system? atom parent?
it appears `New()` was supposed to take xeno_comp, but using `atom` at the parameter appears to be something wrong here